src/memory/memory.ts

```diff
@@ -2,29 +2,54 @@ import { InconsistentMemory, SegmentOutOfBounds } from 'errors/memory';
 
 import { Relocatable } from 'primitives/relocatable';
 import { SegmentValue } from 'primitives/segmentValue';
+import { BuiltinRunner, NoBuiltin } from 'builtins/builtin';
+
+export class Segment {
+  builtin: BuiltinRunner;
+  values: Array<SegmentValue>;
+
+  constructor(builtin: BuiltinRunner = new NoBuiltin()) {
+    this.builtin = builtin;
+    this.values = [];
+  }
+
+  get(address: Relocatable): SegmentValue | undefined {
```

Comment:
`get` should only have an offset and not a `Relocatable` (what would this mean if the segment is not the same?)